### PR TITLE
Switch to go mod and rev version of go-libaudit

### PR DESCRIPTION
**Pull request by andrewstucki**
This does a few things:

1. Switches the project over to using `go mod`
2. Adds semantic versioning for the library
3. Adds a `Vagrantfile` and and `.go-version` similar to beats to aid development on non-linux boxes.

### CODE REVIEW

**Comment by andrewkroh** [comment]
Can you update the changelog too please.

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Read the branch's Go version from the .go-version file.
```
Comment (andrewkroh) [inline Vagrantfile:1]: This is convenient 👍 .

File: .travis.yml [hunk 2]
```diff
@@ -2,31 +2,34 @@
...
 go:
-- 1.10.x
+  - 1.14.x
 go_import_path: github.com/elastic/go-libaudit
 
 install:
-- go get -u github.com/elastic/go-licenser
-- go get -d -t ./...
+  - GO111MODULE=off go get -u github.com/elastic/go-licenser
+  - go get -d -t ./...
```
Comment (andrewkroh) [inline .travis.yml:10]: With go modules I think this command can be removed.

File: .travis.yml [hunk 3]
```diff
@@ -2,31 +2,34 @@
...
-- go build -o build/bin/audit ./cmd/audit/
-- go build -o build/bin/auparse ./cmd/auparse/
+  - go mod download
+  - go mod verify
+  - go mod tidy && [ -z "$(git status --porcelain go.mod go.sum)" ] || (echo "Go module manifest changed. Run 'go mod tidy'" 1>&2 && exit 1)
+  - go-licenser -d
+  - >
+    find . -name '*.go' | grep -v vendor | xargs gofmt -s -l | read &&
+    echo "Code differs from gofmt's style. Run 'gofmt -s -w .'" 1>&2 && exit 1 || true
+  - go test -v $(go list ./... | grep -v /vendor/)
```
Comment (andrewkroh) [inline .travis.yml:20]: Not a new issue so you don't have to change it if you don't want, but the vendor filtering hasn't been needed for a while (there's no `vendor/` code and I think go tools skip vendor when using `./...` now).

**Comment by andrewstucki** [comment]
Sounds good on the `.travis.yml` suggestions I'll add some of the changes in my follow-up PR.

**Pull request merged**